### DISCUSSION TITLE
bad optional access error

### DISCUSSION THREAD

**Discussion by hyeongseok21** [ANLAB-KAIST/KENSv3#92] [Project Q&A 4. Etc]
Hello, we encountered [bad optional access error] which occurs when "attempt to access the value of an object that does not contain a value"  while doing PA4.

We searched through testrouting.cpp file, and added some printf code for debug, and found parameter of netSystem.run() is 3000000000000. 

It seems C++ error, how can I solve this?

here is my debug code and error message indicating value 3000000000000 is inserted automatically

![image](https://user-images.githubusercontent.com/66310634/173292227-7b7f0f76-3312-4a5b-996f-fc265ffffab6.png)
![image](https://user-images.githubusercontent.com/66310634/173293563-e801c7c8-27f1-42ba-874f-719411ce93b9.png)

**Comment by Pusnow** [comment]
> Hello, we encountered [bad optional access error] which occurs when "attempt to access the value of an object that does not contain a value" while doing PA4.

This is related to [std::optional](https://en.cppreference.com/w/cpp/utility/optional). Please, refer to the documentation for its usage.

> We searched through testrouting.cpp file, and added some printf code for debug, and found parameter of netSystem.run() is 3000000000000.

3,000,000,000,000 is a correct value. 3,000 seconds is 3,000,000,000,000 nanoseconds

**Reply by hyeongseok21** [reply]
I've read std::optional documentation several times, but could not find a way to solve the error yet...
I just found that I need either catch (std::bad_optional_access const&) or catch (std::exception const&) for dealing with the exception but still I cannot get a situation when does "attempt to access the value of an object that does not contain a value" happens in my code.

Sorry but could you elaborate more about when does that situation happen?

**Reply by Pusnow** [reply]
You should figure out which optional value is incurring the exception.

It might be helpful to add `--gtest_catch_exceptions=0` argumrnt while executing test binary.
